flang/lib/Lower/Bridge.cpp

```diff
@@ -1945,11 +1942,9 @@ class FirConverter : public Fortran::lower::AbstractConverter {
     auto store = storeMap.find(aliasOffset)->second;
     auto i8Ty = builder->getIntegerType(8);
     auto i8Ptr = builder->getRefType(i8Ty);
-    auto seqTy = builder->getRefType(builder->getVarLenSeqTy(i8Ty));
-    auto base = builder->createConvert(loc, seqTy, store);
     llvm::SmallVector<mlir::Value, 1> offs{
         builder->createIntegerConstant(loc, idxTy, sym.offset() - aliasOffset)};
-    auto ptr = builder->create<fir::CoordinateOp>(loc, i8Ptr, base, offs);
+    auto ptr = builder->create<fir::CoordinateOp>(loc, i8Ptr, store, offs);
```

Comment:
So we're guaranteed to never see another `ref<char>` now?

Reply:
We are guaranteed that aggregate stores are `fir.ref<fir.array<size x type>>`.